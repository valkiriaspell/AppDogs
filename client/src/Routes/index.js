import React from "react";
import { Route, Switch } from "react-router";
import Dogs from "../Component/dogs.jsx";
import DogById from "../Component/dogById.jsx";
import Nav from "../Component/Nav.jsx"
import LandingPage from "../Component/LandingPage.jsx"
import CreateDog from "../Component/CreateDog.jsx"




const index = ({ dog }) => {
    // console.log(dog)
    return(
            <div className="App">
                
                <Route exact path="/">
                    <LandingPage> </LandingPage> 
                </Route>
                <Route path='/home' component={Nav} />
                <Route exact path="/home"> 
                <Dogs></Dogs></Route>
                <Route path="/home/:id" component={DogById}/>
                <Route path='/home/createDog'> <CreateDog></CreateDog> </Route>
                    {/* <Route exact path='/main' />
                    <Route path='/main/create_dog' />
                    <Route path='/main/detail/:id' /> */}
                
            </div>
    );
};
export default index;