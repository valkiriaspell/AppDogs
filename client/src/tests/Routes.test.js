
import Routes from '../Routes/index';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from "react";
import { Route } from "react-router";



describe("<Routes />", () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<Routes />);
    });
  
    it('deber√≠a mostrar <Routes /> correctamente ', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("Deberia renderizar 6 <Route />", () => {
      expect(wrapper.find(Route)).toHaveLength(6);
    });

    it('La primera ruta debe llevar hacia "/" ', () => {
      
      expect(wrapper.find(Route).at(0).prop("path")).toEqual("/");   
      
    });

    it('La segunda ruta debe llevar hacia "/home" ', () => {
      expect(wrapper.find(Route).at(1).prop("path")).toEqual("/home");      
       
    });
  });