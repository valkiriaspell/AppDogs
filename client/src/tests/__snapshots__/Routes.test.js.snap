// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes /> deber√≠a mostrar <Routes /> correctamente  1`] = `
<div
  className="App"
>
  <img
    alt="homedog"
    className="Home_dog"
    src="dogBall.gif"
  />
  <Route
    exact={true}
    path="/"
  >
    <LandingPage>
       
    </LandingPage>
  </Route>
  <Route
    component={[Function]}
    path="/home"
  />
  <Route
    component={[Function]}
    path="/createDog"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/home/:id"
  />
  <Route
    exact={true}
    path="/home"
  >
    <Dogs>
       
    </Dogs>
  </Route>
  <Route
    path="/createDog"
  >
     
    <CreateDog />
     
  </Route>
</div>
`;
